jsk_arc2017_baxter/euslisp/lib/baxter-interface.l

```diff
@@ -0,0 +1,208 @@
+  (:get-arm-controller (arm)
+    (cond ((eq arm :rarm) :rarm-controller)
+          ((eq arm :larm) :larm-controller)
+          (t nil)))
+  (:start-grasp
+    (&optional (arm :arms))
+    (dolist (l/r (if (eq arm :arms) (list "left" "right") (list (arm2str arm))))
+      (if (ros::get-param "/apc_on_gazebo" nil)
+        (ros::service-call
+          (format nil "/robot/~a_vacuum_gripper/on" l/r)
+          (instance std_srvs::EmptyRequest :init))
+        (ros::publish
+          (format nil "/vacuum_gripper/limb/~a" l/r)
+          (instance std_msgs::Bool :init :data t)))))
+  (:stop-grasp
+    (&optional (arm :arms))
+    (dolist (l/r (if (eq arm :arms) (list "left" "right") (list (arm2str arm))))
+      (if (ros::get-param "/apc_on_gazebo" nil)
+        (ros::service-call
+          (format nil "/robot/~a_vacuum_gripper/off" l/r)
+          (instance std_srvs::EmptyRequest :init))
+        (ros::publish
+          (format nil "/vacuum_gripper/limb/~a" l/r)
+          (instance std_msgs::Bool :init :data nil)))))
+  (:gripper-servo-on
+    (&optional (arm :arms))
+    (dolist (l/r (if (eq arm :arms) (list "left" "right") (list (arm2str arm))))
+      (ros::publish
+        (format nil "/gripper_front/limb/~a/servo/torque" l/r)
+        (instance std_msgs::Bool :init :data t))))
+  (:gripper-servo-off
+    (&optional (arm :arms))
+    (dolist (l/r (if (eq arm :arms) (list "left" "right") (list (arm2str arm))))
+      (ros::publish
+        (format nil "/gripper_front/limb/~a/servo/torque" l/r)
+        (instance std_msgs::Bool :init :data nil))))
+  (:graspingp
+    (arm)
+    (let (topic)
+      (if (ros::get-param "/apc_on_gazebo" nil)
+        (progn
+          (setq topic (format nil "/robot/~a_vacuum_gripper/grasping" (arm-to-str arm)))
+          (send (one-shot-subscribe topic std_msgs::Bool) :data)
+          )
+        (progn
+          (setq topic (format nil "gripper_front/limb/~a/pressure/state" (arm-to-str arm)))
+          (< (send (one-shot-subscribe topic std_msgs::Float64) :data)
+             (cond
+               ((eq arm :rarm) rarm-pressure-threshold-)
+               ((eq arm :larm) larm-pressure-threshold-)))))))
+  (:arm-potentio-vector
+    (arm)
+    (case arm
+      (:larm (subseq (send *ri* :state :potentio-vector) 1 9))
+      (:rarm (subseq (send *ri* :state :potentio-vector) 9 17))
+      (t nil)))
+  (:tf-pose->coords
+    (frame_id pose)
+    (let ((base-name (send (send (send self :robot) :base_lk) :name)))
+      (send (send tfl- :lookup-transform base-name frame_id (ros::time 0))
+            :transform (ros::tf-pose->coords pose))))
+  (:fold-pose-back
+    (&optional (arm :arms))
+    (dolist (tmp-arm (if (eq arm :arms) (list :rarm :larm) (list arm)))
+      (send (send self :robot) :fold-pose-back tmp-arm))
+    (send self :send-av))
+  (:send-av
+    (&optional (tm 3000) (ctype nil))
+    (send self :angle-vector (send *baxter* :angle-vector) tm ctype))
```

Comment:
Now what do you think about `:fold-pose-back` and `:send-av`?

I think those methods are sometimes very convenient, but very odd methods because they depend on `*baxter*`
(You rewrite `:fold-pose-back`, but this will cause that `*baxter*` never be reset to `fold-pose-back`)
I think we should stop using those methods or move them to `arc-interface`.